### PR TITLE
[DEV-4663] align sdb models to migrations

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
Run CI with pull_request event here
https://github.com/athenianco/athenian-api/runs/7547122214?check_suite_focus=true
https://github.com/athenianco/athenian-api/runs/7574761678?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -68,6 +68,22 @@ jobs:
     name: custom checks
     if: "!contains(github.event.head_commit.message, 'Bump version') || github.event_name != 'push'"
     runs-on: ubuntu-22.04
+    services:
+      postgres:
+        image: postgres:11.8
```
Comment (vmarkovtsev) [inline .github/workflows/push.yml:73]: We are using 13 in prod, it is better to test with 13 in the CI

File: server/athenian/api/models/state/env.py [hunk 2]
```diff
@@ -27,6 +29,10 @@
 # ... etc.
 
 
+def _sqlalchemy_url_override() -> Optional[str]:
+    return os.environ.get("ALEMBIC_SDB", None)
```
Comment (vmarkovtsev) [inline server/athenian/api/models/state/env.py:33]: We shouldn't modify the env file. It is trivial to patch the config file with `sed` in the CI as needed.

File: server/athenian/api/models/state/models.py [hunk 3]
```diff
@@ -325,6 +327,7 @@ class MappedJIRAIdentity(create_time_mixin(created_at=True, updated_at=True), Ba
     """JIRA identity mapping."""
 
     __tablename__ = "jira_identity_mapping"
+    __table_args__ = (Index("jira_identity_mapping_jira_user_id", "account_id", "jira_user_id"),)
```
Comment (vmarkovtsev) [inline server/athenian/api/models/state/models.py:330]: No-no, indexes are too complex and advanced to mention in the models. We should ignore their existence.

File: server/scripts/check_sdb_migrations.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+dbname=check_sdb_migrations
+
+export ATHENIAN_INVITATION_KEY=777 # needed by migrations
+
+psql -c "DROP DATABASE IF EXISTS $dbname" -h 0.0.0.0 -p 5432  -d postgres
+psql -c "CREATE DATABASE $dbname TEMPLATE 'template0' LC_COLLATE 'C.UTF-8'" -h 0.0.0.0 -p 5432
+
+python3 -m athenian.api.models.state "postgresql://$PGUSER:$PGPASSWORD@0.0.0.0:5432/$dbname"
+# a spurious backup table is left by migrations
+psql -c "DROP TABLE jira_identity_mapping_old" -h 0.0.0.0 -p 5432  -d $dbname
```
Comment (vmarkovtsev) [inline server/scripts/check_sdb_migrations.sh:16]: We should drop this table in a migration, it is an old artifact.

File: server/scripts/check_sdb_migrations.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+set -eu
+
+echo $PGUSER > /dev/null
+echo $PGPASSWORD > /dev/null
```
Comment (vmarkovtsev) [inline server/scripts/check_sdb_migrations.sh:6]: This doesn't make sense :smile:

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: server/scripts/check_sdb_migrations.sh (hunk 5 above)
Comment (gaetano-guerriero) [reply server/scripts/check_sdb_migrations.sh:6]: This tests that the variables are set, with set -u

File: server/athenian/api/models/state/models.py (hunk 3 above)
Comment (gaetano-guerriero) [reply server/athenian/api/models/state/models.py:330]: I don't understand, should we only manually create them in migrations ? Do we have examples of indexes we were not able to declare in models ? We already have indexes on single columns, unique constraints, check constraints with a custom expression in the models.

Comment (gaetano-guerriero) [reply server/athenian/api/models/state/models.py]: I missed the INCLUDE

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**